app/models/product.rb

```diff
@@ -216,6 +216,21 @@ def uncommitted_files
     git_status
   end
 
+  def readme?
+    found_readme = false
+    self.router.path.each_child do |path|
+      if path.file? && path.readable? && path.basename.to_s =~ /^readme(.*)/i
+        @readme ||= path
+        found_readme = true
+      end
+    end
+    return found_readme
+  end
+
+  def readme
+    return readme? ? @readme : nil
+  end
```

Comment:
This logic and the view logic for rendering the readme would be simpler if there were a separate `ProductReadme` class with methods `html`, `exist?`, `title`, `name`. You can also use the Rails view link and path/url helpers in the class by `include Rails.application.routes.url_helpers` so then you could add a `ProductReadme#edit_link`.

The method `ProductReadme#html` could return the rendered content, whether it is a plain text file `<pre>This README is not markdown formatted</pre>` or a markdown file `<h1>Markdown file</h1> <p>Yeah!</p>`.

For finding the readme you can use `Pathname.glob`:

```
2.2.0 :023 > app
 => #<Pathname:/Users/efranz/dev/ood-dashboard>
2.2.0 :024 > Pathname.glob app.join("readme*"), File::FNM_CASEFOLD
 => [#<Pathname:/Users/efranz/dev/ood-dashboard/README.md>]
```

Since glob always returns an array, you can do this:

```ruby
# returns
def readme_path
  Pathname.glob(app.join("readme*"), File::FNM_CASEFOLD).first
end
```

Of course it returns nil if there is no readme, which is not ideal but is okay in this case if the method is just used internally.

Also, of course, in the view you would still access the readme the same way: `@product.readme`